src/lib/translators/mongo.ts

```diff
@@ -290,6 +290,53 @@ function transformReplace(step: Readonly<S.ReplaceStep>): MongoStep {
   };
 }
 
+/** transform an 'replace' step into corresponding mongo steps */
+function transformRollup(step: Readonly<S.RollupStep>): MongoStep {
+  const facet: { [id: string]: MongoStep[] } = {};
+  for (const [idx, elem] of step.hierarchy.entries()) {
+    const id = columnMap([...step.hierarchy.slice(idx), ...(step.groupby ?? [])]);
+    const aggs: { [id: string]: {} } = {};
+    for (const aggfStep of step.aggregations) {
+      if (aggfStep.aggfunction === 'count') {
+        aggs[aggfStep.newcolumn] = {
+          $sum: 1,
+        };
+      } else {
+        aggs[aggfStep.newcolumn] = {
+          [$$(aggfStep.aggfunction)]: $$(aggfStep.column),
+        };
+      }
+    }
+    const project: { [id: string]: string | number } = {
+      _id: 0,
+      ...Object.fromEntries(Object.keys(id).map(col => [col, `$_id.${col}`])),
+      ...Object.fromEntries(Object.keys(aggs).map(col => [col, 1])),
+      [step.labelCol ?? 'label']: `$_id.${elem}`,
```

Comment:
It's just a matter of style a discoverability, but I'd feel better seeing all the default values declared up in the function, with their default values:
```
const labelCol = step.labelCol ?? 'label';
const levelCol = step.levelCol ?? 'level';
[...]
```
This way, we only have the default values in one place and it's easy to see them all at once.